packages/fxa-settings/src/pages/InlineRecoveryKeySetup/container.tsx

```diff
@@ -3,22 +3,145 @@
+import { getSyncNavigate } from '../Signin/utils';
+import { hardNavigate } from 'fxa-react/lib/utils';
+import { formatRecoveryKey } from '../../lib/utilities';
 
-export const InlineRecoveryKeySetupContainer = ({
-  integration,
-}: {
-  integration: Integration;
-} & RouteComponentProps) => {
+export const InlineRecoveryKeySetupContainer = (_: RouteComponentProps) => {
   const [currentStep, setCurrentStep] = useState<number>(1);
-  const createRecoveryKeyHandler = useCallback(async () => {
-    // TODO in FXA-10079
-  }, []);
+  const [formattedRecoveryKey, setFormattedRecoveryKey] = useState<string>('');
+  const ftlMsgResolver = useFtlMsgResolver();
+  const authClient = useAuthClient();
+
+  const location = useLocation() as ReturnType<typeof useLocation> & {
+    state?: SigninLocationState;
+  };
+  const { email, uid, sessionToken, unwrapBKey } = location.state || {};
+
+  const sensitiveDataClient = useSensitiveDataClient();
+  const sensitiveData = sensitiveDataClient.getData(AUTH_DATA_KEY);
+  const { authPW, emailForAuth } =
+    (sensitiveData as unknown as { emailForAuth: string; authPW: string }) ||
+    {};
+
+  const navigateForward = useCallback(() => {
+    setCurrentStep(currentStep + 1);
+  }, [currentStep]);
+
+  const createRecoveryKey = useCallback(
+    (
+        uid: string,
+        sessionToken: string,
+        unwrapBKey: string
+      ): (() => Promise<CreateRecoveryKeyHandler>) =>
+      async () => {
+        try {
+          // We must reauth for another `keyFetchToken` because we sent it to Sync
+          const reauth = await authClient.sessionReauthWithAuthPW(
+            sessionToken,
+            emailForAuth,
+            authPW,
+            {
+              keys: true,
+              reason: 'recovery_key',
+            }
+          );
+          if (!reauth.keyFetchToken) throw new Error('Invalid keyFetchToken');
+          const keys = await authClient.accountKeys(
+            reauth.keyFetchToken,
+            unwrapBKey
+          );
+          const { recoveryKey, recoveryKeyId, recoveryData } =
+            await generateRecoveryKey(uid, keys);
+          await authClient.createRecoveryKey(
+            sessionToken,
+            recoveryKeyId,
+            recoveryData
+          );
+
+          cache.modify({
+            id: cache.identify({ __typename: 'Account' }),
+            fields: {
+              recoveryKey() {
+                return {
+                  exists: true,
+                };
+              },
+            },
+          });
+          setFormattedRecoveryKey(formatRecoveryKey(recoveryKey.buffer));
+          navigateForward();
+          return { data: { recoveryKey } };
+        } catch (error) {
+          return {
+            localizedErrorMessage: ftlMsgResolver.getMsg(
+              'inline-recovery-key-setup-create-error',
+              'Oops! We couldn’t create your recovery key. Please try again later.'
```

Comment:
I'm changing this to `account recovery key` per Valerie asking / Eduardo confirming with my next push.